src/pip/_internal/pyproject.py

```diff
@@ -33,13 +34,18 @@ def make_pyproject_path(unpacked_source_directory):
     return path
 
 
+BuildsystemDetails = namedtuple('BuildsystemDetails', [
+    'requires', 'backend', 'check', 'backend_path'
```

Comment:
By default these will have type `Any` ([link](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#named-tuples)). I don't know a Python 2 compatible way less verbose than a class, with type-annotated `__init__`, to express this safely.

Reply:
Are you asking for this to be a class? I found a handful of other namedtuples in the code, so I assumed that was OK.

I understand why type annotations are useful. But I feel like the stricter we are about applying them everywhere, the more we step away from the simplicity that made Python useful in the first place.

Reply:
Not necessarily, just calling attention to it (and hoping I was wrong and someone would mention a succinct way to do it).

In general we should at least try to maintain the amount of typed code. One reason this is important in the pip code base is because our tests take much much longer to run than the type checking so it's really helpful as part of a quick feedback loop during development.

In this case since we're only using the return value in one place and unpacking it immediately (and I can't see having more than one caller in the future or passing this value around), I would leave the return type a `Tuple`.

Reply:
@uranusjr are you OK with the argument of reverting to a plain tuple to preserve the type hints?

Reply:
I’m okay with that (but maybe we’ll need to start considering what should happen if `[build-system]` continues to grow).

Reply:
Personally, I'd prefer sticking with a namedtuple.

I'm fine with type annotations adding value in the form of code checking, but I'm not comfortable if we have to modify the way we write code to cater for typing limitations.

> I understand why type annotations are useful. But I feel like the stricter we are about applying them everywhere, the more we step away from the simplicity that made Python useful in the first place.

Precisely this.

Reply:
In general I disagree, motivated by a feeling that types and type checking help me get more done with fewer bugs and less effort. That said, this is easy to change later so I won't block this PR over it.